### ISSUE TITLE
Define contribution types

### ISSUE THREAD

**Issue by adamj89** [Green-Software-Foundation/speakers#61]

**Comment by zanete** [comment]
Capturing the status of the weekly call:
- @adamj89 as there has not been any movement on this since last week, we need to **emphasise** that this has to be finalised ASAP (by the 15th of September as the latest) as the data formats need to be final before any development starts. Updating it half way development will lead to complexities we can't afford. The timing is not on our side, the development has a high risk of being late and jeopardise the launch at Decarb. 

cc @jawache

**Comment by adamj89** [comment]
I’m on leave today. This will be updated by Friday (in time for Monday) as per the project plan.

Best wishes,
Adam Jackson
Program Manager - Green Software Foundation
Community Working Group

**Comment by zanete** [comment]
- we are assuming that a champion could be more than one type (e.g a Speaker and a Writer or mentor)
- we can compute the type judging by the type of activity they add
- we can simplify the data structure to make it more generic for all contribution types. we propose:
- - (required) contribution type (speaking, writing etc)
- - (optional) contribution-specific subtype (e.g.  which Article type or Talk type) - for points, assume "Other" when not provided
- - (required) date from - used for sorting chronologically
- - (optional) date to (for a date range)
- - (required) title
- - (optional) short description
- - (required) URL 
- Phase 1 (for Decarb launch) is not going to include automatic scoring, that will be added in phase II
- will create another issue to discuss the solution for flagging experts and storing scores for each champion

**Comment by adamj89** [comment]
Yes - we will encourage multiple contribution types. Thank you for this summary.

**Comment by jawache** [comment]
One other thing I realised that will ensure the contributions align to our goals and needs is that every contribution should be linked to one of our projects and/or working groups. So another attribute is a link or name of the project/WG this contribution supports.

We can also have a process where in the project/WG calls we give the group the chance to object to any claim of contribution. Most contributions are going to be honest, some are going to be questionable and some are just going to be people gaming the system to gain a badge. 

So if you make an OSS contribution, it should be something that is linked one way or another to one of our projects. Doesn't need to be code submitted to one of our repos, you might be adding support for carbon aware sdk to another oss project owned by someone else but you must be linked to something in the GSF.

I think that's a lightweight way to ensure we have some governance and oversight and also gives pressure to ensure the program is driving contributions to our projects.

**Comment by adamj89** [comment]
Great point! We will add this. 

I think we need to keep it optional (at least for speaking, writing) as folks in Champions (and speakers) are not all GSF members. For meetup organisers we would attach to the meetup project by default I guess. It can definitely be mandatory for some categories. 

I will update this structure to reflect this @osamajandali

**Comment by adamj89** [comment]
Have added linked project as a mandatory field for all contribution types. We can add guidance notes for speakers talking about a general topic to link to CWG.

**Comment by zanete** [comment]
@adamj89 will beef it up with the Champion entity params

**Comment by adamj89** [comment]
@osamajandali I have split location into

Country
City

Both required
